pkg/support/dump_test.go

```diff
@@ -38,3 +100,279 @@ func TestParseFileName(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, ts.String(), "2021-08-17 09:47:58 +0000 UTC")
 }
+
+func TestTimestampFilter(t *testing.T) {
+	ts := timestampFilter("7s")
+	currentTime := time.Now()
+	assert.Contains(t, ts.String(), currentTime.Format("2006-01-02"))
+}
+
+func TestWriteFile(t *testing.T) {
+	tests := []struct {
+		name           string
+		basedir        string
+		resource       string
+		data           []byte
+		expectedWinErr string
+		expectedErr    string
+	}{
+		{
+			name:     "writefile with specifying data",
+			resource: "networkpolicies",
+			basedir:  "dir1/dir2",
+			data:     []byte("antct -oyaml get networkpolicies"),
+		},
+		{
+			name:           "writefile without specifying data",
+			expectedWinErr: "error when writing  to file: open : The system cannot find the file specified.",
+			expectedErr:    "error when writing  to file: open : no such file or directory",
```

Comment:
I am not sure we really need to treat the Linux & Windows cases differently. Instead we should just use `assert. ErrorContains` and match the largest common substring. Same for other tests.